BASE/src/Microsoft.ApplicationInsights/Metrics/Extensibility/MetricValuesBuffer.cs

```diff
@@ -91,13 +93,18 @@ public TValue GetAndResetValue(int index)
                 while (this.IsInvalidValue(value))
                 {
                     spinWait.SpinOnce();
-                    
+
                     if (spinWait.Count % 100 == 0)
                     {
                         // In tests (including stress tests) we always finished wating before 100 cycles.
                         // However, this is a protection against en extreme case on a slow machine. 
                         Task.Delay(10).ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
                     }
+                    else if (spinWait.Count > 10000)
+                    {
+                        // exceeded maximum spin count. Break out to avoid infinite loop.
+                        return this.InvalidValue;
```

Comment:
can we simply return `value`?

Reply:
yes, because it's already an invalid value. 🤦‍♂️
good catch!